x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -181,17 +186,22 @@ class AgentPolicyService {
       updated_by: user ? user.username : 'system',
     });
 
+    const newAgentPolicy = await this.get(soClient, id, false);
+
+    newAgentPolicy!.package_policies = existingAgentPolicy.package_policies;
+
     if (options.bumpRevision || options.removeProtection) {
       await this.triggerAgentPolicyUpdatedEvent(esClient, 'updated', id, {
         spaceId: soClient.getCurrentNamespace(),
+        agentPolicy: newAgentPolicy,
       });
     }
     logger.debug(
       `Agent policy ${id} update completed, revision: ${
         options.bumpRevision ? existingAgentPolicy.revision + 1 : existingAgentPolicy.revision
       }`
     );
-    return (await this.get(soClient, id)) as AgentPolicy;
+    return newAgentPolicy as AgentPolicy;
```

Comment:
The agent policy could be updated through the code `triggerAgentPolicyUpdatedEvent` so refetching it is probably one the easiest way to get the updated policy

Reply:
@nchaulet yeah i think that's a good point, in many cases though return isn't even being used in case of bumpRevision. I guess i can revert this or make it conditional.

Reply:
Yes maybe introducing a new option `returnUpdatedPolicy` that default to true could work, but it will change this interface and a lot of related interface, not sure if it's worth the change

Reply:
i have handled it